### ISSUE TITLE
Speed up threading.Condition wakeup

### ISSUE THREAD

**Issue by 84401114-8e59-4056-83cb-632106c0b648** [python/cpython#59344]
BPO | [15139](https://bugs.python.org/issue15139)
--- | :---
Nosy | @tim-one, @loewis, @birkenfeld, @gpshead, @jcea, @pitrou, @kristjanvalur, @anacrolix
Files | <li>[condition.patch](https://bugs.python.org/file26085/condition.patch "Uploaded as text/plain at 2012-06-22.15:38:28 by @kristjanvalur")</li><li>[queuetest.py](https://bugs.python.org/file26086/queuetest.py "Uploaded as text/plain at 2012-06-22.15:38:56 by @kristjanvalur"): Test file to measure wakeup delay</li><li>[condition.patch](https://bugs.python.org/file26117/condition.patch "Uploaded as text/plain at 2012-06-23.21:19:15 by @kristjanvalur")</li><li>[condition.patch](https://bugs.python.org/file26243/condition.patch "Uploaded as text/plain at 2012-07-03.11:03:14 by @kristjanvalur")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2012-06-22.15:38:31.754>
labels = ['interpreter-core', 'library', 'performance']
title = 'Speed up threading.Condition wakeup'
updated_at = <Date 2014-04-04.14:18:20.388>
user = 'https://github.com/kristjanvalur'
```

bugs.python.org fields:
```python
activity = <Date 2014-04-04.14:18:20.388>
actor = 'pitrou'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Interpreter Core', 'Library (Lib)']
creation = <Date 2012-06-22.15:38:31.754>
creator = 'kristjan.jonsson'
dependencies = []
files = ['26085', '26086', '26117', '26243']
hgrepos = []
issue_num = 15139
keywords = ['patch']
message_count = 17.0
messages = ['163423', '163426', '163427', '163428', '163436', '163592', '163597', '163659', '163660', '163664', '163676', '163681', '164347', '164590', '164647', '164650', '215520']
nosy_count = 10.0
nosy_names = ['tim.peters', 'loewis', 'georg.brandl', 'gregory.p.smith', 'jcea', 'pitrou', 'kristjan.jonsson', 'anacrolix', 'sbt', 'Catalin.Patulea']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'performance'
url = 'https://bugs.python.org/issue15139'
versions = ['Python 3.4']
```

</p></details>

**Comment by iritkatriel** [comment]
This has been idle for about a decade now. Is it still relevant?

**Comment by gpshead** [comment]
Our `threading.Condition` class is still pure Python, but the RLock (default) behind it is implemented in native code these days, unlike in 2.7. I don't doubt there's room for improvement but i'm inclined to close this, someone can reopen it or open a new issue if they want to improve the throughput of our threading synchronization related APIs.
